nodecg-io-serial/extension/SerialClient.ts

```diff
@@ -0,0 +1,92 @@
+import { ServiceClient } from "nodecg-io-core/extension/types";
+import { success, error, Result } from "nodecg-io-core/extension/utils/result";
+import { ReadLine } from "readline";
+import SerialPort = require("serialport"); // This is neccesary, because serialport only likes require!
+// This is neccesary because serialport does not work with import...
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const Readline = require("@serialport/parser-readline");
+
+export interface DeviceInfo {
+    port: string;
+    manucaturer: string;
+    serialNumber: string;
+    pnpId: string;
+}
+
+interface Protocoll {
+    delimiter: string;
+    encoding: string;
```

Comment:
If you use `SerialPort.parsers.Readline` as I above recommended the compiler will complain because encoding has to be one off these strings instead of any string. You may also want to update this in the json schema.
```suggestion
    encoding: 'ascii' | 'utf8' | 'utf16le' | 'ucs2' | 'base64' | 'binary' | 'hex';
```

Reply:
Whoops, I originally only added the most common encodings like ascii, utf8, binary and hex in the schema and did not think about the interface... 
I totally agree with you.